source/architecture.rst

```diff
@@ -1,230 +1,245 @@
-making possible a multi-tenant architecture for the middleware.
+This service is stateless, having its data persisted to a database, with data
+isolation for users and applications, making possible a multi-tenant
+architecture for the middleware.
 
-Iot-agent
+IoT Agent
 *********
 
-The iot-agent is an adaptation service between the data model of the context broker and the devices data models. The
-*dojot* platform can have multiple iot-agents, each one of them being specialized in a specific protocol like, for
-instance, MQTT/JSON, CoAP/LWM2M and HTTP/JSON.
+An IoT agent is an adaptation service between physical devices and *dojot* core
+components. It could be understood as a *device driver* for a set of devices.
+The *dojot* platform can have multiple iot-agents, each one of them being
+specialized in a specific protocol like, for instance, MQTT/JSON, CoAP/LWM2M
+and HTTP/JSON.
+
+It is also responsible for all security mechanisms used in communication
+channels between the devices it deals with and the component itself.
 
-Security measures like the management of the secure channel used for the communication of the platform with the device
-is also treated by this service.
 
 User Authorization Service
 **************************
 
-This service is responsible for managing user profiles and access control. Basically any API call that reaches the
-platform via the API Gateway is validated by this service.
+This service is responsible for managing user profiles and access control.
+Basically any API call that reaches the platform via the API Gateway is
+validated by this service.
 
-To be able to deal with a high volume of authorization calls, it uses caching, it is stateless and it is scalable
-horizontally. Its data is stored on a database.
+To be able to deal with a high volume of authorization calls, it uses caching,
+it is stateless and it is scalable horizontally. Its data is stored on a
+database.
 
-Service Orchestrator
-********************
+flowbroker
+**********
 
-This service provides a high level API for configuring the *dojot* with the objective of reducing the need of knowing
-how to handle each one of the services that are part of the platform. More specifically, it is responsible for modeling
-different services, exposing a simplified configuration interface and propagating this configuration to the various
-services when requested. It acts as a centralized configuration manager for multiple services.
-
-Fiware Perseo
-**************
+This service provides a high level API for configuring the *dojot* with the
+objective of reducing the need of knowing how to handle each one of the
+services that are part of the platform. More specifically, it is responsible
+for modeling different services, exposing a simplified configuration interface
+and propagating this configuration to the various services when requested. It
+acts as a centralized configuration manager for multiple services.
```

Comment:
Calling those flow "action blocks" as "services" mixes their concept with the other services the platform provides making this whole statement very confusing. All the other platform blocks have been called "services" so far, so the flow nodes need a different way to be referenced

Reply:
I removed this part.